File: WorkManager/WorkManager.Data.Models/Enums/HistoryModificationType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace WorkManager.Data.Enums
{
    [DataContract]
    public enum HistoryModificationType : byte
    {
        [EnumMember]
        Insert = 0,
        [EnumMember]
        Update = 1,
        [EnumMember]
        Delete = 2
    }
}

File: WorkManager.Server/src/WorkManager/Data/Enums/HistoryModificationType.java
package WorkManager.Data.Enums;

import java.util.HashMap;
import java.util.Map;

public enum HistoryModificationType
{
        Insert((byte)0),
        Update((byte)1),
        Delete((byte)2);

        private byte value;
        private static Map map = new HashMap<>();

        private HistoryModificationType(byte value) {
                this.value = value;
        }

        static {
                for (HistoryModificationType pageType : HistoryModificationType.values()) {
                        map.put(pageType.value, pageType);
                }
        }

        public static HistoryModificationType valueOf(byte pageType) {
                return (HistoryModificationType) map.get(pageType);
        }

        public int getValue() {
                return value;
        }
}
